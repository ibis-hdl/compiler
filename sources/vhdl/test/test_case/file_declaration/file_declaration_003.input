file F3: IntegerFile open WRITE_MODE is "test.dat";

