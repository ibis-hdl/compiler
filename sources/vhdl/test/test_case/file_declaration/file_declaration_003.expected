(file_declaration
  (identifier_list
    (identifier
      F3
    )
  )
  (subtype_indication
    (name<v>
      (identifier
        IntegerFile
      )
    )
  )
  (file_open_information
    (expression
      (relation
        (shift_expression
          (simple_expression
            (term
              (primary<v>
                (name<v>
                  (identifier
                    WRITE_MODE
                  )
                )
              )
            )
          )
        )
      )
    )
    (expression
      (relation
        (shift_expression
          (simple_expression
            (term
              (primary<v>
                (literal<v>
                  (string_literal
                    test.dat
                  )
                )
              )
            )
          )
        )
      )
    )
  )
)
